### PR TITLE
First pass on video drawer

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
(Required)

#### What's this PR do?
(Required)

#### How should this be manually tested?
- Reimport youtube videos (`course_catalog.etl.pipelines.youtube_etl()`)
- Go to search page and verify the drawer works and data displays as expected (e.g. duration, etc)

#### Screenshots (if appropriate)

Desktop:
Screenshot_2019-11-08 MIT Open Learning

Mobile:
Screenshot_2019-11-08 MIT Open Learning(2)

### CODE REVIEW

**Comment by pdpinch** [comment]
Looks good, but we have an updated design in InVision: https://projects.invisionapp.com/d/main#/console/17477251/390731424/preview

**Comment by rhysyngsun** [comment]
@pdpinch if you're referring the styling differences, I discussed with @abdulkdawson friday about that, the conclusion out of that was that new styling for all of the drawer designs is out of scope for this work since that styling is shared across all the drawer types, so this PR focuses on the functional display of the information relevant to video types and the playback of the video.

If you're referring to the partial height of the drawer, I also discussed that with Abdul since it's a departure from our other drawer designs. Some preliminary testing of mine indicated that this will cause some overall issues with scrolling of the sidebar, particularly on mobile. My opinion is that this is likely to be fussy and delicate at best (on top of the drawer already being fussy and delicate). I don't think there was a decision for that particular design aspect so I left this as-is until I heard otherwise.

**Comment by pdpinch** [comment]
I was referring primarily to the first issue — thank you for checking with Abdul about that. 

Am I correct in assuming, then, that you’re not making any significant changes to the styling in this PR?

**Comment by rhysyngsun** [comment]
@pdpinch there are no styling changes at all - the player embed didn't need any adjustments so the default styling already defined for the drawer is what we're getting here.

**Comment by aliceriot** [comment]
works great!

my one question is if we want to make it easier to open the video in a new tab - just wondering because some of these videos are quite long and watching a one-hour video from the drawer seems a bit fragile - like it might be frustrating for the user to get 15 minutes in and then accidentally click outside the drawer and lose their progress.

idk, just something that occurred to me, what do you think @abdulkdawson ?

File: static/js/lib/learning_resources.js [hunk 1]
```diff
@@ -228,3 +231,44 @@ export const filterItems = (
...
 export const privacyIcon = (privacyLevel: string) =>
   privacyLevel === "public" ? "lock_open" : "lock"
+
+export const isCoursewareResource = R.contains(R.__, [
+  LR_TYPE_COURSE,
+  LR_TYPE_PROGRAM,
+  LR_TYPE_BOOTCAMP
+])
+
+export const formatDurationClockTime = (value: string) => {
```
Comment (aliceriot) [inline static/js/lib/learning_resources.js]: seems like something moment should support itself!

**Comment by abdulkdawson** [comment]
@aliceriot I completely agree with you on this one. I may have to run this past Ferdi and Peter.

File: static/js/lib/learning_resources.js (hunk 1 above)
Comment (rhysyngsun) [reply static/js/lib/learning_resources.js]: yeah, there's a closed issue on their repo, which people keep commenting on that it's not really what they needed: https://github.com/moment/moment/issues/463

**Pull request merged**